crates/sui-node/src/lib.rs

```diff
@@ -1764,17 +1764,27 @@ impl SuiNode {
                 tokio::time::sleep(Duration::from_millis(1)).await;
 
                 let config = cur_epoch_store.protocol_config();
-                let binary_config = to_binary_config(config);
+                let chain_id = cur_epoch_store.get_chain();
+                let supported_greater_versions = self
+                    .config
+                    .supported_protocol_versions
+                    .expect("Supported versions should be populated")
+                    // no need to send digests of versions less than the current version
+                    .truncate_below(config.version);
+
+                // Pick the latest binary config and use that to determine the available system packages.
+                let binary_config = {
+                    let latest_protocol_config =
+                        ProtocolConfig::get_for_version(supported_greater_versions.max, chain_id);
+                    to_binary_config(&latest_protocol_config)
```

Comment:
If I understand correctly, if there is a mix of highest supported versions on the network, and the binary config is different between those two versions, then this will result in the network not being able to upgrade, even though an upgrade would be possible if all nodes used the binary config of the lower of the two versions. Is this correct?